cmd/boots/main.go

```diff
@@ -221,14 +221,19 @@ func main() {
 	}
 }
 
-func getFinders(l log.Logger, c *config) (client.WorkflowFinder, client.HardwareFinder, error) {
+func getFinders(l log.Logger, c *config, reporter client.Reporter) (client.WorkflowFinder, client.HardwareFinder, error) {
 	var hf client.HardwareFinder
 	var wf client.WorkflowFinder = &client.NoOpWorkflowFinder{}
 	var err error
 
 	switch os.Getenv("DATA_MODEL_VERSION") {
 	case "":
-		hf, err = cacher.NewHardwareFinder(os.Getenv("FACILITY_CODE"))
+		p, ok := reporter.(cacher.Poster)
```

Comment:
Rather than expose the `Poster` interface and do this type assertion, can you just make `getDiscoveryFromEM` a private method on the `cacher.HardwareFinder`?

Reply:
the issue is that getDiscoveryEM needs a way to talk to emapi, I don't really want to duplicate bits of Reporter just for that, and I need a way to plumb down a client or creds through cacher.NewHardwareFinder so that it can be used if ByMAC needs it.

Its all ugly, we're just going back and forth on the least ugly route. I'm kinda going back to `Reporter.Post` not being that bad considering all of `Reporter` only ever really makes sense in cacher mode anyway.

Reply:
Oh I get it, I was conflating `Finder` with `Reporter`. Then I'd probably opt for `Reporter.Post()` and avoid the type assertion

Reply:
:D yeah. So going back to v1.

Reply:
sorry for the churn!